reactioncommerce/reaction
# [WIP] Fix 3821: currency component not converting currency

Resolves #3821 
Impact: **major**  
Type: **bugfix**

## Issue
The issue was basically that the component was using the user selected currency for conversion.

## Solution
The conversion function now uses the shop's base currency
Also fixed some issues like
1. Only number can be entered in the currecny
1. There is a separate reusable CurrencyInput component now.

## Testing
1. As an admin, set the shop base currency to EUR.
1. Edit a variant of the `Example Product`
1. Input a price
1. The value should not get converted to any other value as soon as the field blurs. Only currency formatting should be added.
1. Try entering a non-digit character in the input box. 
1. Notice you are unable to do so.
